examples/metrics/projection.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: 0BSD
+"""
+Understanding Data Projection
+=============================
+
+Projecting data onto P3 definitions.
+
+The P3 Analysis Library expects data to be prepared in a specific
+:ref:`format <performance_data>`. This format was inspired by the
+:ref:`terminology <terminology>` first introduced in "`Implications of a Metric
+for Performance Portability`_":
+
+.. _Implications of a Metric for Performance Portability:
+   https://doi.org/10.1016/j.future.2017.08.007
+
+**Problem**
+  A task with a pass/fail metric for which quantitative performance may be
+  measured.
+
+**Application**
+  Software capable of solving a *problem* with measurable correctness and
+  performance.
+
+**Platform**
+  A collection of software and hardware on which an *application* may run a
+  *problem*.
+
+These definitions are flexible, allowing the same performance data to be used
+for multiple case studies with different interpretations of these terms.
+
+Rather than store raw performance data in columns corresponding to these
+definitions, the P3 Analysis Library provides functionality to *project* raw
+performance data onto specific meanings of "problem", "application" and
+"platform".
+
+Using Projection to Rename Columns
+----------------------------------
+
+The simplest example of projection is a straightforward renaming of columns.
+
+Let's assume that we've collected some performance data from a few different
+implementations of a function, running a number of problem sizes on multiple
+machines.
+
+.. important::
+    Although we are looking at "function" performance here, the concepts
+    generalize to entire software packages.
+
+Our raw performance data might look like this:
+
+.. list-table::
+    :widths: 20 20 20 20
+    :header-rows: 1
+
+    * - size
```

Comment:
suggest this is titled "problem size", rather than just "size"

Reply:
I initially had this as "problem size", but I removed the "problem" to make the projection clearer.